django/db/models/functions/base.py

```diff
@@ -25,6 +25,9 @@ def __init__(self, expression, output_field):
     def as_sql(self, compiler, connection, **extra_context):
         if 'db_type' not in extra_context:
             extra_context['db_type'] = self.output_field.db_type(connection)
+        # Support CharField casting without max_length.
```

Comment:
Yes, but I think that we haven't reached consensus in the [discussion](https://groups.google.com/forum/#!topic/django-developers/h1vMu_k0JcA/discussion). Moreover, in some backends proposed data type is appropriate for casting without precision but cannot be used as a column data type e.g. `char` on MySQL. In other backends we need to use `TextField` data type e.g. `text` on SQLite. Only PostgreSQL has this specific feature that you can use both `varchar` and `varchar(max_length)` as a column data type.

Maybe I should rename `DatabaseWrapper.data_type_char_field_without_max_length`?